lib/web_ui/lib/src/engine/semantics/semantics_helper.dart

```diff
@@ -0,0 +1,316 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+part of engine;
+
+/// The maximum [semanticsActivationAttempts] before we give up waiting for
+/// the user to enable semantics.
+///
+/// This number is arbitrary and can be adjusted if it doesn't work well.
+const int kMaxSemanticsActivationAttempts = 20;
+
+/// The message in the label for the placeholder element used to enable
+/// accessibility.
+///
+/// This uses US English as the default message. Set this value prior to
+/// calling `runApp` to translate to another language.
+String placeholderMessage = 'Enable accessibility';
+
+class SemanticsHelper {
+  final _EnableSemantics _enableSemantics =
+      isDesktop ? DesktopEnableSemantics() : MobileEnableSemantics();
+
+  bool shouldEnableSemantics(html.Event event) {
+    return _enableSemantics.tryEnableSemantics(event);
+  }
+
+  html.Element prepareAccesibilityPlaceholder() {
+    return _enableSemantics.prepareAccesibilityPlaceholder();
+  }
+}
+
+abstract class _EnableSemantics {
+  /// A temporary placeholder used to capture a request to activate semantics.
+  html.Element _semanticsPlaceholder;
+
+  /// We do not immediately enable semantics when the user requests it, but
+  /// instead wait for a short period of time before doing it. This is because
+  /// the request comes as a tap on the [_semanticsPlaceholder]. The tap,
+  /// depending on the browser, comes as a burst of events. For example, Safari
+  /// sends "touchstart", "touchend", and "click". So during a short time period
+  /// we consume all events and prevent forwarding to the framework. Otherwise,
+  /// the events will be interpreted twice, once as a request to activate
+  /// semantics, and a second time by Flutter's gesture recognizers.
+  @visibleForTesting
+  Timer semanticsActivationTimer;
```

Comment:
This class should only contain things that are called through the `_EnableSemantics` type. Everything else should be inside the implementation class.

Reply:
done.

i am addressing this comments. 

on the other hand i believe this suggestion reduces the common usage of code increases the class size. we will always need semanticsActivationTimer and _semanticsPlaceholder. i also don't think we will be adding another form factor soon.